tests/agent/actions/checkpoints/test_run_checkpoint_action.py

```diff
@@ -19,85 +23,61 @@
-    return RunCheckpointEvent(
-        type="run_checkpoint_request",
-        checkpoint_id=checkpoint_id,
-        datasource_names_to_asset_names=datasource_names_to_asset_names,
-    )
-
-
-@pytest.fixture
-def checkpoint_event_with_splitter_options(checkpoint_id, datasource_names_to_asset_names):
-    return RunCheckpointEvent(
-        type="run_checkpoint_request",
-        checkpoint_id=checkpoint_id,
-        datasource_names_to_asset_names=datasource_names_to_asset_names,
-        splitter_options={"year": 2023, "month": 11, "day": 30},
-    )
-
-
-def test_run_checkpoint_action_without_splitter_options_returns_action_result(
-    mock_context, checkpoint_event_without_splitter_options, checkpoint_id
-):
-    action = RunCheckpointAction(context=mock_context)
-    id = "096ce840-7aa8-45d1-9e64-2833948f4ae8"
-    checkpoint = mock_context.run_checkpoint.return_value
-    checkpoint.ge_cloud_id = checkpoint_id
-    checkpoint.run_results = {
-        "GXCloudIdentifier::validation_result::78ebf58e-bdb5-4d79-88d5-79bae19bf7d0": {
-            "actions_results": {
-                "store_validation_result": {"id": "78ebf58e-bdb5-4d79-88d5-79bae19bf7d0"}
-            }
-        }
-    }
-    action_result = action.run(event=checkpoint_event_without_splitter_options, id=id)
-
-    mock_context.run_checkpoint.assert_called_with(
-        ge_cloud_id=UUID(checkpoint_id), batch_request=None
-    )
-    assert action_result.type == checkpoint_event_without_splitter_options.type
-    assert action_result.id == id
-    assert action_result.created_resources == [
-        CreatedResource(
-            resource_id="78ebf58e-bdb5-4d79-88d5-79bae19bf7d0",
-            type="SuiteValidationResult",
+@pytest.mark.parametrize(
+    "splitter_options, batch_request",
+    [
+        (
+            {"year": 2023, "month": 11, "day": 30},
+            {"options": {"day": 30, "month": 11, "year": 2023}},
         ),
-    ]
-
-
-def test_run_checkpoint_action_with_splitter_options_returns_action_result(
-    mock_context, checkpoint_event_with_splitter_options, checkpoint_id
+        ({}, None),
+        (None, None),
+    ],
+)
+@pytest.mark.parametrize(
+    "action_class,event",
+    [
+        (
+            RunCheckpointAction,
+            RunCheckpointEvent(
+                type="run_checkpoint_request",
+                datasource_names_to_asset_names={"Data Source 1": {"Data Asset A", "Data Asset B"}},
+                checkpoint_id="5f3814d6-a2e2-40f9-ba75-87ddf485c3a8",
+            ),
+        ),
+        (
+            RunScheduledCheckpointAction,
+            RunScheduledCheckpointEvent(
+                type="run_scheduled_checkpoint.received",
+                datasource_names_to_asset_names={"Data Source 1": {"Data Asset A", "Data Asset B"}},
+                checkpoint_id="5f3814d6-a2e2-40f9-ba75-87ddf485c3a8",
+                schedule_id=UUID("5f3814d6-a2e2-40f9-ba75-87ddf485c3a8"),
+            ),
```

Comment:
Avoiding a duplicate set of tests by using parametrization in this test file for both the scheduled and non-scheduled version.

Reply:
`non-blocking`

Are you using the exact same parameters here and for `test_run_checkpoint_action_raises_on_test_connection_failure`?

If so you put them both in the same [test class](https://docs.pytest.org/en/8.2.x/getting-started.html#group-multiple-tests-in-a-class) and put the parameters at the class level.

Reply:
Oooh good call. Yep - I can clean up either in this PR or a followup.

Reply:
Not finding a great way to reference as class level variables in the pytest parametrization decorator. I moved them to the module level at least to DRY it up a bit. Open to suggestions for a followup PR if there is a cleaner way.